text/0000-react-global-hooks.md

```diff
@@ -0,0 +1,135 @@
+- **Start Date:** (fill me in with today’s date, 2022-05-22
+- **RFC PR:** (leave this empty)
+
+# `@xstate/react` Global Hooks
+
+## Summary
+
+This RFC seeks to establish a simple way to create hooks for global actors (instances of XState machines) that can be accessed in any component. This will allow for shared state patterns, similar to Redux, Zustand, Recoil, and others, without the need for React context providers. Shared state is a very common use-case for non-trivial React applications, and this would enable state machines to be used at any level: component and/or app-wide (global) state with ease.
+
+## Motivation
+
+TODO
+
+Inspiration:
+
+- [Zustand](https://github.com/pmndrs/zustand)
+- [Redux (in React)](https://react-redux.js.org/)
+- [Recoil](https://recoiljs.org/)
+
+## Detailed design
+
+**Quick overview:**
+```js
+import { createHooks } from '@xstate/react';
```

Comment:
bikeshedding: `bindHooks`?

Reply:
`withHooks` implies the decorator pattern.

Reply:
I chose `createHooks` to stay consistent with the `create*` pattern elsewhere in XState; happy to bikeshed and get other opinions though

Reply:
It just feels to me that we are not creating much here - we just grab existing functions (hooks) and prebind them with the given argument (actorRef). I don't feel super strong about it though.